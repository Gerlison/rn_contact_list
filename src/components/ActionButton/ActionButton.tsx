import React, { memo } from 'react';
import { ImageSourcePropType, TouchableOpacityProps } from 'react-native';
import styled from 'styled-components/native';

interface Props extends TouchableOpacityProps {
  icon: ImageSourcePropType;
}

const ActionButton = ({ icon, ...props }: Props): JSX.Element => {
  return (
    <S.Touchable testID="action button" {...props}>
      <S.Image source={icon} />
    </S.Touchable>
  );
};

const S = {
  Touchable: styled.TouchableOpacity`
    align-items: center;
    justify-content: center;
    padding: 18px;
    border-radius: 100px;
    background-color: #7c37a6;
    position: absolute;
    bottom: 32px;
    right: 18px;
  `,
  Image: styled.Image.attrs({
    resizeMode: 'contain',
  })`
    width: 22px;
    height: 22px;
  `,
};

export default memo(ActionButton);
