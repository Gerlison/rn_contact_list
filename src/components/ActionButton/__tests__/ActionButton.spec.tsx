import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import ActionButton from '../ActionButton';

import icPlus from '../../../../assets/icons/icPlus.png';

describe('ActionButton', () => {
  it('SHOULD match snapshot', () => {
    const component = render(<ActionButton icon={icPlus} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('SHOULD call onPress on click', () => {
    const mockFn = jest.fn();
    const component = render(<ActionButton icon={icPlus} onPress={mockFn} />);
    fireEvent.press(component.getByTestId('action button'));
    expect(mockFn).toBeCalled();
  });
});
