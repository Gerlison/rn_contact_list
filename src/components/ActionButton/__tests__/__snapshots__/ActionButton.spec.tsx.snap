// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton SHOULD match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7c37a6",
      "borderRadius": 100,
      "bottom": 32,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingLeft": 18,
      "paddingRight": 18,
      "paddingTop": 18,
      "position": "absolute",
      "right": 18,
    }
  }
  testID="action button"
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Array [
        Object {
          "height": 22,
          "width": 22,
        },
      ]
    }
  />
</View>
`;
